### ISSUE TITLE
False positive on `no-return-wrap`

### ISSUE THREAD

**Issue by ammut** [eslint-community/eslint-plugin-promise#661]
```ts
type SetupWithTeardown = () => Promise<() => Promise<void>>;
const foo: SetupWithTeardown = () => {
  // do some sync work...
  return Promise.resolve(() => Promise.resolve()); // error: no-return-wrap
}
```

This errors for me with v7.2.1. I don't think this should be an error, because it's just another way of writing this:

```ts
const foo: SetupWithTeardown = async () => {
  // do some sync work...
  return async () => { /* nothing */ };
}
```

And not flagging something where `Promise.resolve` is not actually needed.
